File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.java
class Solution {
    public int[] searchRange(int[] nums, int target) {
      int first=ffind(nums,target);
      int last=lfind(nums,target);
      return new int[]{first,last};
    }
    public int ffind(int[] nums,int target){
        int l=0;
        int r=nums.length-1;
        int res=-1;
        while(l<=r){
            int mid=l+(r-l)/2;
            if(nums[mid]==target){
                res=mid;
                r=mid-1;
            }
            else if(nums[mid]<target){
                l=mid+1;
            }
            else{
                r=mid-1;
            }
        }
        return res;
    }
    public int lfind(int[] nums,int target){
        int l=0;
        int r=nums.length-1;
        int res=-1;
        while(l<=r){
            int mid=l+(r-l)/2;
            if(nums[mid]==target){
                res=mid;
                l=mid+1;
            }
            else if(nums[mid]<target){
                l=mid+1;
            }
            else{
                r=mid-1;
            }
        }
        return res;
    }
}

File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.py
class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        def ffind(nums,tar):
            l=0
            r=len(nums)-1
            res=-1
            while l<=r:
                m=l+(r-l)//2
                if nums[m]==tar:
                    res=m
                    r=m-1
                elif nums[m]<=tar:
                    l=m+1
                else:
                    r=m-1
            return res
        def lfind(nums,tar):
            l=0
            r=len(nums)-1
            res=-1
            while l<=r:
                m=l+(r-l)//2
                if nums[m]==tar:
                    res=m
                    l=m+1
                elif nums[m]<tar:
                    l=m+1
                else:
                    r=m-1
            return res
        return [ffind(nums,target),lfind(nums,target)]